params/config.go

```diff
@@ -67,6 +73,17 @@ var (
 		MaxBlockGasCost:  big.NewInt(1_000_000),
 		BlockGasCostStep: big.NewInt(200_000),
 	}
+
+	// For UpgradeConfig Marshal/Unmarshal
+	magicHeader              = []byte{0xff}
+	sectionHeaderSizes       = 1
+	precompileHeader         = []byte{0xaa}
+	stateUpdateHeader        = []byte{0xc8}
+	endHeader                = []byte{0}
```

Comment:
Do we need these headers? I believe these are typically used to signal what type of data they include, but we should know that when serializing/de-serializing, so I don't think these should be necessary.

Reply:
Fixed in 8935259f, instead I'm having a deterministic serialization